### PR TITLE
Build a better utility for making local stock data

### PR DESCRIPTION
**Pull request by esoergel**
@emord

### CODE REVIEW

**Comment by emord** [comment]
:+1: after tests

File: corehq/apps/commtrack/tests/util.py [hunk 1]
```diff
@@ -217,16 +224,30 @@ def get_single_transfer_block(src_id, dest_id, product_id, quantity, date_string
...
-    Don't use this with a real user
-    """
-    if not user.phone_number:
-        raise ValueError("User does not have a phone number")
-    if not user.get_verified_number():
-        user.save_verified_number(user.domain, user.phone_number, True, None)
-    domain_obj = Domain.get_by_name(user.domain)
-    parser = StockReportParser(domain_obj, user.get_verified_number())
-    process(user.domain, parser.parse(text.lower()))
+@require_debug_true()
```
Comment (snopoke) [inline corehq/apps/commtrack/tests/util.py]: Is there a reason you couldn't use the `unit_testing_only` decorator (https://github.com/dimagi/commcare-hq/blob/4a25e003b8e2deb6417a11f21cf46ad21a47737e/corehq/util/test_utils.py#L25-L25)

Comment (esoergel) [inline corehq/apps/commtrack/tests/util.py]: I made this for using locally, not during tests, to generate stock data for reports.

Comment (snopoke) [inline corehq/apps/commtrack/tests/util.py]: 👍

On 17 December 2015 at 11:38, Ethan Soergel <notifications@github.com>
wrote:

> In corehq/apps/commtrack/tests/util.py
> <https://github.com/dimagi/commcare-hq/pull/9608#discussion_r47885596>:
>
> > @@ -217,16 +224,30 @@ def get_single_transfer_block(src_id, dest_id, product_id, quantity, date_string
> >      ).strip()
> >
> >
> > -def fake_sms(user, text):
> > -    """
> > -    Fake a commtrack SMS submission for a user.
> > -    `text` might be "soh myproduct 100"
> > -    Don't use this with a real user
> > -    """
> > -    if not user.phone_number:
> > -        raise ValueError("User does not have a phone number")
> > -    if not user.get_verified_number():
> > -        user.save_verified_number(user.domain, user.phone_number, True, None)
> > -    domain_obj = Domain.get_by_name(user.domain)
> > -    parser = StockReportParser(domain_obj, user.get_verified_number())
> > -    process(user.domain, parser.parse(text.lower()))
> > +@require_debug_true()
>
> I made this for using locally, not during tests, to generate stock data
> for reports.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/dimagi/commcare-hq/pull/9608/files#r47885596>.
>



-- 
Simon Kelly
Senior Engineer | Dimagi South Africa

**Pull request merged**